### PR TITLE
radio value coming as string in on change handlers

### PR DESCRIPTION
**Pull request by kakanisridhar**
# Description
Fix for issue https://github.com/XGovFormBuilder/digital-form-builder/issues/149 
Go to "Form details"
Click on "yes"
Then you can't click on "no"

previous to this fix as the event handler is not setting state correctly radio buttons get struck and  do not respond

## Aceptance criteria. 

Given I am on "Form details" screen
Then I should be able to switch between "Yes" / "No" multiple times without any issues

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

- [x] I have added tests that prove my changes are effective 
- [x] New and existing unit tests pass locally with my changes

# Checklist:

- [x] My changes do not introduce any new linting errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and versioning
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have rebased onto master and there are no code conflicts

### CODE REVIEW

**Review by jenbutongit — APPROVED** [review]
since this is an open source repository, could you describe the fix and add acceptance criteria if necessary please. not everyone will have access to our jira instance

**Review by bertrandmc — CHANGES_REQUESTED** [review]

File: designer/client/helpers.js
Comment (kakanisridhar) [reply designer/client/helpers.js:74]: ok

File: designer/test/form-details.test.tsx
Comment (kakanisridhar) [reply designer/test/form-details.test.tsx:2]: i was playing around with them as Radios is not mounting with all its content , so i resorted to testing the method, will remove

File: designer/client/form-details.js
Comment (jenbutongit) [inline designer/client/form-details.js]: `isFeedbackForm = e.target.value === "true"` was nicer imo. Since there are only two options true or false, you only have to evaluate once. the str2bool function is only helpful if you are going to be casting a bunch of string values.

File: designer/client/form-details.js
Comment (jenbutongit) [inline designer/client/form-details.js:58]: I know this wasn't your code, so you don't have to action it but would be nice to refactor here. (Feel free to dismiss this review comment, I wont block merge over it!) 

we can remove the entire if else block and just do this 
`this.setState({ feedbackForm: isFeedbackForm, selectedFeedbackForm: !isFeedbackForm ? undefined : this.state.selectedFeedbackForm  });`

File: designer/client/helpers.js
Comment (jenbutongit) [inline designer/client/helpers.js:1]: are you using this? delete if not

File: designer/client/helpers.js
Comment (jenbutongit) [inline designer/client/helpers.js]: see comment above. Also, this code probably doesn't do what you're expecting it to. I would assume str2bool will return a boolean, but it can return a string if it is not 'true' or 'false'

File: designer/test/form-details.test.tsx
Comment (jenbutongit) [inline designer/test/form-details.test.tsx]: remove any of the methods you're not using

**Review by jenbutongit — CHANGES_REQUESTED** [review]

File: designer/client/form-details.js
Comment (kakanisridhar) [reply designer/client/form-details.js]: we are passing boolean values to Radios, not sure why we are getting string values, so this function is just catering for future changes, if the Radios pass back boolean values

**Review by bertrandmc — APPROVED** [review]
LGTM

**Review by jenbutongit — APPROVED** [review]

**Pull request merged**
